src/Microsoft.DotNet.Build.Tasks/Targets/depending.targets

```diff
@@ -1,18 +1,34 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <Target Name="EnsureDependencies">
-    <ItemGroup>
-      <PackageConfigs Include="$(SourceDir)**\packages.config" />
-    </ItemGroup>
+  <PropertyGroup>
+    <RestoreAllPackageDependenciesSemaphoreFile>$(PackagesDir)\RestoreAllPackageDependencies.complete</RestoreAllPackageDependenciesSemaphoreFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageConfigs Include="$(SourceDir)**\packages.config" Exclude="@(PackageConfigsToExclude)" />
+  </ItemGroup>
+
+  <!-- Keep this target around until we can update depending repros to use the new name -->
+  <Target Name="EnsureDependencies" DependsOnTargets="RestoreAllPackageDependencies" />
+
+  <Target Name="RestoreAllPackageDependencies"
+    Inputs="@(PackageConfigs)"
+    Outputs="$(RestoreAllPackageDependenciesSemaphoreFile)"
+    DependsOnTargets="$(RestoreAllPackageDependenciesDependsOn)"
+    Condition="'@(PackageConfigs)'!=''"
+    >
+    <Message Importance="High" Text="Restoring all package dependencies..." />
+    
+    <Error Condition="'$(PackagesDir)' == ''" Text="PackagesDir property needs to be set before calling RestoreAllPackageDependencies targets" />
 
     <!-- Restore Packages -->
     <Exec
-      Condition="'@(PackageConfigs)'!=''"
       StandardOutputImportance="Low"
       Command="&quot;$(NuGetToolPath)&quot; restore &quot;%(PackageConfigs.FullPath)&quot; -PackagesDirectory &quot;$(PackagesDir.TrimEnd('\'))&quot; $(NuGetConfigCommandLine)" 
       />
 
+    <Touch Files="$(RestoreAllPackageDependenciesSemaphoreFile)" AlwaysCreate="true" />
```

Comment:
I still don't understand why we need this. I've seen this in other targets as well.